extensions/resource-deployment/src/ui/deployClusterWizard/pages/deploymentProfilePage.ts

```diff
@@ -0,0 +1,214 @@
+		this.wizard.model.setPropertyValue(VariableNames.ControllerDataStorageSize_VariableName, selectedProfile.controllerDataStorageSize);
+		this.wizard.model.setPropertyValue(VariableNames.ControllerLogsStorageSize_VariableName, selectedProfile.controllerLogsStorageSize);
+		this.wizard.model.setPropertyValue(VariableNames.EnableHADR_VariableName, selectedProfile.hadrEnabled);
+		this.wizard.model.setPropertyValue(VariableNames.SQLServerPort_VariableName, selectedProfile.sqlServerPort);
+		this.wizard.model.setPropertyValue(VariableNames.GateWayPort_VariableName, selectedProfile.gatewayPort);
+		this.wizard.model.setPropertyValue(VariableNames.ControllerPort_VariableName, selectedProfile.controllerPort);
+		this.wizard.model.setPropertyValue(VariableNames.IncludeSpark_VariableName, selectedProfile.includeSpark);
+		this.wizard.model.setPropertyValue(VariableNames.ControllerDataStorageClassName_VariableName, selectedProfile.controllerDataStorageClass);
+		this.wizard.model.setPropertyValue(VariableNames.ControllerLogsStorageClassName_VariableName, selectedProfile.controllerLogsStorageClass);
+		this.wizard.model.setPropertyValue(VariableNames.ReadableSecondaryPort_VariableName, selectedProfile.sqlServerReadableSecondaryPort);
+		this.wizard.model.adAuthSupported = selectedProfile.activeDirectorySupported;
+		this.wizard.model.selectedProfile = selectedProfile;
+	}
+
+	private loadCards(): Thenable<void> {
+		return this.wizard.azdataService.getDeploymentProfiles().then((profiles: BigDataClusterDeploymentProfile[]) => {
+			const defaultProfile: string = this.getDefaultProfile();
+
+			profiles.forEach(profile => {
+				const card = this.createProfileCard(profile, this._view!);
+				if (profile.profileName === defaultProfile) {
+					card.selected = true;
+					this.setModelValuesByProfile(profile);
+				}
+				this._cardContainer!.addItem(card, { flex: '0 0 auto' });
+			});
+		});
+	}
+
+	public onEnter() {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			this.wizard.wizardObject.message = { text: '' };
+			if (pcInfo.newPage > pcInfo.lastPage) {
+				const isValid = this.wizard.model.getStringValue(VariableNames.DeploymentProfile_VariableName) !== undefined;
+				if (!isValid) {
+					this.wizard.wizardObject.message = {
+						text: localize('deployCluster.ProfileNotSelectedError', "Please select a deployment profile."),
+						level: azdata.window.MessageLevel.Error
+					};
+				}
+				return isValid;
+			}
+			return true;
+		});
+	}
+
+	public onLeave() {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	private getDefaultProfile(): string {
+		switch (this.wizard.deploymentType) {
+			case BdcDeploymentType.NewAKS:
+			case BdcDeploymentType.ExistingAKS:
+				return 'aks-dev-test';
+			case BdcDeploymentType.ExistingKubeAdm:
+				return 'kubeadm-dev-test';
+			default:
+				return 'aks=dev-test';
+		}
```

Comment:
ask-dev-test? This should probably be removed and an error thrown if no valid deployment type is selected though right?

Reply:
right.